Src/DependencyCollector/Net46.Tests/DependencyTrackingTelemetryModuleTestNet46.cs

```diff
@@ -117,28 +129,42 @@ public void TestDependencyCollectionEventSource404()
                 module.Initialize(this.config);
                 Assert.IsFalse(DependencyTableStore.IsDesktopHttpDiagnosticSourceActivated);
 
-                var url = new Uri("http://google.com/404");
                 HttpClient httpClient = new HttpClient();
-                httpClient.GetStringAsync(url).ContinueWith(t => { }).Wait();
+                using (new LocalServer(
+                    LocalhostUrl, 
+                    ctx => 
+                    {
+                        // https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/548
+                        // for quick unsuccesfull response OnEnd is fired too fast after begin (before it's completed)
+                        // first begin may take a while because of lazy initializations and jit compiling
+                        // let's wait a bit here.
+                        Thread.Sleep(20);
+                        ctx.Response.StatusCode = 404;
+                    }))
+                {
+                    await httpClient.GetAsync(LocalhostUrl).ContinueWith(t => { });
+                }
 
-                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, null, false, "404");
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), new Uri(LocalhostUrl), null, false, "404");
             }
         }
 
         [TestMethod]
         [Timeout(5000)]
-        public void TestDependencyCollectionDiagnosticSource404()
+        public async Task TestDependencyCollectionDiagnosticSource404()
         {
             using (var module = new DependencyTrackingTelemetryModule())
             {
                 module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
                 module.Initialize(this.config);
 
-                var url = new Uri("http://google.com/404");
                 HttpClient httpClient = new HttpClient();
-                httpClient.GetStringAsync(url).ContinueWith(t => { }).Wait();
+                using (new LocalServer(LocalhostUrl, ctx => ctx.Response.StatusCode = 404))
+                {
+                    await httpClient.GetStringAsync(LocalhostUrl).ContinueWith(t => { });
+                }
 
-                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), url, null, false, "404");
+                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), new Uri(LocalhostUrl), null, false, "404");
```

Comment:
Note that for 404 (where the response is available) DiagnosticSource tracks the end. Event source tracks dependency only for dns or other network issue (when there is no response). I'll update corresponding test